src/Avataaars/Random.elm

```diff
@@ -0,0 +1,304 @@
+        , Avataaars.Eyes.Dizzy
+        , Avataaars.Eyes.EyeRoll
+        , Avataaars.Eyes.Happy
+        , Avataaars.Eyes.Hearts
+        , Avataaars.Eyes.Side
+        , Avataaars.Eyes.Squint
+        , Avataaars.Eyes.Surprised
+        , Avataaars.Eyes.Wink
+        , Avataaars.Eyes.WinkWacky
+        ]
+
+
+generateMouth : Generator Mouth
+generateMouth =
+    Random.uniform Avataaars.Mouth.Default
+        [ Avataaars.Mouth.Concerned
+        , Avataaars.Mouth.Disbelief
+        , Avataaars.Mouth.Eating
+        , Avataaars.Mouth.Grimace
+        , Avataaars.Mouth.Sad
+        , Avataaars.Mouth.ScreamOpen
+        , Avataaars.Mouth.Serious
+        , Avataaars.Mouth.Smile
+        , Avataaars.Mouth.Tongue
+        , Avataaars.Mouth.Twinkle
+        , Avataaars.Mouth.Vomit
+        ]
+
+
+generateFace : Generator Face
+generateFace =
+    Random.map3 Face generateMouth generateEyes generateEyebrow
+
+
+generateFacialHair : Generator FacialHair
+generateFacialHair =
+    generateHairColor
+        |> Random.andThen
+            (\color ->
+                Random.uniform Avataaars.FacialHair.Blank
+                    [ Avataaars.FacialHair.BeardMedium color
+                    , Avataaars.FacialHair.BeardLight color
+                    , Avataaars.FacialHair.BeardMajestic color
+                    , Avataaars.FacialHair.MoustacheFancy color
+                    , Avataaars.FacialHair.MoustacheMagnum color
+                    ]
+            )
+
+
+generateHairColor : Generator HairColor
+generateHairColor =
+    Random.uniform Avataaars.HairColor.auburn
+        [ Avataaars.HairColor.black
+        , Avataaars.HairColor.blonde
+        , Avataaars.HairColor.blondeGolden
+        , Avataaars.HairColor.brown
+        , Avataaars.HairColor.brownDark
+        , Avataaars.HairColor.pastelPink
+        , Avataaars.HairColor.platinum
+        , Avataaars.HairColor.red
+        , Avataaars.HairColor.silverGray
+        ]
+
+
+generateHatColor : Generator HatColor
+generateHatColor =
+    Random.uniform Avataaars.HatColor.black
+        [ Avataaars.HatColor.blue01
+        , Avataaars.HatColor.blue02
+        , Avataaars.HatColor.blue03
+        , Avataaars.HatColor.gray01
+        , Avataaars.HatColor.gray02
+        , Avataaars.HatColor.heather
+        , Avataaars.HatColor.pastelBlue
+        , Avataaars.HatColor.pastelGreen
+        , Avataaars.HatColor.pastelOrange
+        , Avataaars.HatColor.pastelRed
+        , Avataaars.HatColor.pastelYellow
+        , Avataaars.HatColor.pink
+        , Avataaars.HatColor.red
+        , Avataaars.HatColor.white
+        ]
+
+
+generateSkinTone : Generator SkinTone
+generateSkinTone =
+    Random.uniform Avataaars.SkinTone.black
+        [ Avataaars.SkinTone.brown
+        , Avataaars.SkinTone.darkBrown
+        , Avataaars.SkinTone.light
+        , Avataaars.SkinTone.pale
+        , Avataaars.SkinTone.tanned
+        , Avataaars.SkinTone.yellow
+        ]
+
+
+generateTop : Generator Top
+generateTop =
```

Comment:
The most complex `Generator` since each variant has differing type constructor requirements.

Reply:
You can use `Random.Extra.choices` to simplify it

Reply:
@miniBill I don't know if this is a simpler method, after talking with @wolfadex [in the Elm Slack](https://elmlang.slack.com/archives/C0CJ3SBBM/p1701362324720349), I don't know that `choices` is a "cleaner" route, it seems semantic more than anything from my understanding. WDYT?

Reply:
I also don't know for this use case how the implementation could be modelled correctly since `choices`, in my understanding, would choose one of the 6 `Generator` instances we generate but I need all 6 to cover all of the variants, right?

Reply:
Pushed a version that uses `choices `in this way, let me know if it's what you were looking for.